### DISCUSSION TITLE
Vuepress 2 - inherit default theme but replace a component

### DISCUSSION THREAD

**Discussion by wachunga** [vuepress/core#372] [Q&A] [answered]
Is it possible to extend @vuepress/theme-default but override specific components in Vuepress 2?

I see in the docs it's possible to change layouts and insert things into existing slots, but now I would like to replace the [Sidebar](https://github.com/vuepress/vuepress-next/blob/main/packages/%40vuepress/theme-default/src/client/components/Sidebar.vue) component entirely. Is that possible?

**Comment by Mister-Hope** [comment, marked as answer]
See https://v2.vuepress.vuejs.org/reference/default-theme/extending.html#layout-slots
